Tasks/AzureResourceGroupDeployment/operations/MachineGroupExtensionHelper.ts

```diff
@@ -42,24 +42,26 @@ export class MachineGroupExtensionHelper {
         var listOfVms: az.VM[] = await this.azureUtils.getVMDetails();
         var deleteExtensionFromVmPromises: Promise<any>[] = [];
         for (var vm of listOfVms) {
-            deleteExtensionFromVmPromises.push(this.deleteExtension(vm));
+            deleteExtensionFromVmPromises.push(this.deleteExtensionFromSingleVM(vm));
         }
         await Promise.all(deleteExtensionFromVmPromises);
         if (listOfVms.length > 0) {
             console.log(tl.loc("MGAgentDeletedFromAllVMs"));
         }
     }
 
-    public deleteExtension(vm: az.VM): Promise<any> {
+    public deleteExtensionFromSingleVM(vm: az.VM): Promise<any> {
         return new Promise((resolve, reject) => {
-            var vmName = vm["name"]
-            var resourceGroupName = this.taskParameters.resourceGroupName;
+            var vmName = vm["name"];
             var extensionParameters = this.formExtensionParameters(vm, "delete");
             var extensionName = extensionParameters["extensionName"];
             console.log(tl.loc("DeleteExtension", extensionName, vmName));
-            this.computeClient.virtualMachineExtensions.deleteMethod(resourceGroupName, vmName, extensionName, (error, result, request, response) => {
+            this.computeClient.virtualMachineExtensions.deleteMethod(this.taskParameters.resourceGroupName, vmName, extensionName, (error, result, request, response) => {
                 if (error) {
-                    reject(tl.loc("DeletionFailed", vmName, utils.getError(error)));
+                    tl.loc("DeletionFailed", vmName, utils.getError(error));
```

Comment:
this one should be the message with reject. the other messages you can print as warnings